### PR TITLE
docs: localnet upgrade guide

### PR DESCRIPTION
**Pull request by cosmic-vagabond**
## Description

updated the localnet upgrade guide

### What has Changed?

What specific problem were you aiming to address, and how did you successfully resolve it? If tests were not uploaded for this pull request or if coverage decreased, please provide an explanation for the change.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/elys-network/elys/blob/main/docs/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#testing)
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### Deployment Notes

Are there any specific considerations to take into account when deploying these changes? This may include new dependencies, scripts that need to be executed, or any aspects that can only be evaluated in a deployed environment.

### Screenshots and Videos

Please provide any relevant before and after screenshots by uploading them here. Additionally, demo videos can be highly beneficial in demonstrating the process.

### CODE REVIEW

File: scripts/upgrade-assure/README.md [hunk 1]
```diff
@@ -1,75 +1,132 @@
...
+git tag v0.31.0
 make install
-go run ./scripts/upgrade-assure/... https://snapshots.polkachu.com/testnet-snapshots/elys/elys_5724942.tar.lz4 ~/go/bin/elysd ~/go/bin/elysd --skip-proposal
 ```
 
-## AviaOne testnet snapshots
+### Step 3: Retrieve the Current Binary for Darwin Users
 
-You can find the latest AviaOne testnet snapshot [here](https://aviaone.com/blockchains-service/elystestnet-1-elys.html#8).
+Download the specific binary for Darwin ARM64 architecture using `curl`:
```
Comment (Wgil) [inline scripts/upgrade-assure/README.md:36]: Then here we could say "Download the specific binary using curl

File: scripts/upgrade-assure/README.md [hunk 2]
```diff
@@ -1,75 +1,132 @@
...
-go run ./scripts/upgrade-assure/... --home /tmp/elys --home2 /tmp/elysd2 https://tools.highstakes.ch/files/elys.tar.gz /tmp/elysd-v0.29.30 /tmp/elysd-v0.29.31 --skip-snapshot --skip-chain-init
-```
-# Troubleshooting
 
-These are some problems and its solutions when the script is executed
+#### 6b: Handle Potential Errors
+
+Address any type errors, such as difficulties in unmarshaling strings into integer fields in Go struct fields.
+
+#### 6c: Update the Script
```
Comment (Wgil) [inline scripts/upgrade-assure/README.md:72]: Is 6c step intended to fix errors faced on **6b** right? It would be nice you have a couple of examples here as well on what are the data structures that we need to keep an eye on, what's the path for those and what's the process to update them. Is this related to proto files? Is it an automated process?

Thank you!

File: scripts/upgrade-assure/README.md [hunk 3]
```diff
@@ -1,75 +1,132 @@
...
+Switch back to the main branch to incorporate the latest changes:
+
+```bash
+git checkout main
 ```
--  Timeout, if you have timeout problems with the node or waiting for the next block you can use these flags: 
-timeout-wait-for-node is expressed in seconds
-timeout-next-block is expressed in minutes
 
+### Step 8: Final Upgrade Command
```
Comment (Wgil) [inline scripts/upgrade-assure/README.md:92]: Should we include here the steps to run both alice and bob nodes?

**Review by Wgil — COMMENTED** [review]
Here are some improvement opportunities, thanks!

File: scripts/upgrade-assure/README.md (hunk 3 above)
Comment (cosmic-vagabond) [reply scripts/upgrade-assure/README.md:92]: @Wgil both alice and bob nodes are running automatically when running this command

File: scripts/upgrade-assure/README.md (hunk 2 above)
Comment (cosmic-vagabond) [reply scripts/upgrade-assure/README.md:72]: this will depends on what the error will be returned at the export step, I dont have an example right now but I would be happy to add one in subsequent updates when facing a new instance

File: scripts/upgrade-assure/README.md [hunk 4]
```diff
@@ -1,75 +1,132 @@
...
+
+```bash
+go run ./scripts/upgrade-assure/... /tmp/snapshot.tar.lz4 /tmp/elysd-v0.29.31 ~/go/bin/elysd --skip-node-start
 ```
-go run ./scripts/upgrade-assure/... --home /tmp/elys --home2 /tmp/elysd2 https://tools.highstakes.ch/files/elys.tar.gz /tmp/elysd-v0.29.30 /tmp/elysd-v0.29.31 --skip-snapshot --skip-chain-init
-```
-# Troubleshooting
 
-These are some problems and its solutions when the script is executed
+#### 6b: Handle Potential Errors
```
[the remark this answers is not in the document]
Comment (cosmic-vagabond) [reply scripts/upgrade-assure/README.md:68]: same as below comment

**Comment by Wgil** [comment]
@cosmic-vagabond As part of the step 6 could you please explain what types should exactly be updated? what are the specific folder containing them, if this is a manual task or automated, etc.

I was under the impression that this refactor was going to introduce a lot of type changes but you only removed one field here:

https://github.com/elys-network/elys/pull/435/files#diff-db67b28e480cae78b2e5a3e55372f235502dea0e921e0d0e068c57e143da2b81

Screenshot 2024-04-26 at 10 22 18 AM


If you find this hard to explain because it depends on specific migration, you could comment what's your thought process to approach this and maybe we can work together explaining it.

**Comment by cosmic-vagabond** [comment]
> @cosmic-vagabond As part of the step 6 could you please explain what types should exactly be updated? what are the specific folder containing them, if this is a manual task or automated, etc.
> 
> I was under the impression that this refactor was going to introduce a lot of type changes but you only removed one field here:
> 
> https://github.com/elys-network/elys/pull/435/files#diff-db67b28e480cae78b2e5a3e55372f235502dea0e921e0d0e068c57e143da2b81
> 
> Screenshot 2024-04-26 at 10 22 18 AM
> 
> If you find this hard to explain because it depends on specific migration, you could comment what's your thought process to approach this and maybe we can work together explaining it.

@Wgil `types.go` uses the elys data structure types to convert to JSON the genesis state used to create localnet, but for some structure fields the ones that are using integer as a type, go struggles to convert them, that’s why you can see them overriden to `JSON.Number`, that’s pretty much all this file purpose is for.

Whenever the export step in upgrade-assure fails to parse the genesis JSON file it will return an error and as said it is mostly because of some integer fields that needs to be redefined to JSON.Number.

**Review by Wgil — APPROVED** [review]
Thanks for the explanation, lets include that and we're good to go.
